src/main/java/com/moorhuhnservice/moorhuhnservice/controller/ConfigController.java

```diff
@@ -0,0 +1,66 @@
+package com.moorhuhnservice.moorhuhnservice.controller;
+
+import com.moorhuhnservice.moorhuhnservice.data.*;
+import com.moorhuhnservice.moorhuhnservice.data.mapper.ConfigurationMapper;
+import com.moorhuhnservice.moorhuhnservice.data.mapper.QuestionMapper;
+import com.moorhuhnservice.moorhuhnservice.repositories.ConfigurationRepository;
+import com.moorhuhnservice.moorhuhnservice.service.ConfigService;
+import java.util.List;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("api/v1/minigames/moorhuhn/configurations")
+@Slf4j
+public class ConfigController {
+
+  @Autowired
+  ConfigurationRepository configurationRepository;
+
+  @Autowired
+  ConfigService configService;
+
+  @Autowired
+  QuestionMapper questionMapper;
+
+  @Autowired
+  ConfigurationMapper configurationMapper;
+
+  @GetMapping("")
+  public List<Configuration> getConfigurations() {
+    log.debug("get all configurations");
+    List<Configuration> configurations = configurationRepository.findAll();
+    return configurations;
+  }
+
+  @GetMapping("/{configurationName}")
+  public Configuration getConfiguration(@PathVariable String configurationName) {
+    log.debug("get configuration {}", configurationName);
+    return configService.getConfiguration(configurationName);
+  }
+
+  @GetMapping("/{configurationName}/questions")
```

Comment:
why is the parameter beforr the static name... is there any purpose? i think its ugly

Reply:
You have to think that before is `/configurations`, so the actual endpoint is `/api/v1/minigames/moorhuhn/configurations/{configurationName}/questions`. So it makes more sense instead of `.../configurations/questions/{configurationName}`. You have to take a look to the **Configuration** class.

Reply:
ok then we let it be like you have it